File: programs-basic-java/src/test/java/com/local/numbertypes/NelsonNumberTest.java
package com.local.numbertypes;

import static com.local.numbertypes.NumberType.NELSON_NUMBER;

import java.util.List;

import org.junit.jupiter.api.Test;

class NelsonNumberTest extends AbstractNumberTest {

	@Test
	void testNelsonNumbers() {
		testNumbersTypeByFile(NELSON_NUMBER, "nelsonnumbers.txt");
	}

	@Test
	void testNelsonNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(NELSON_NUMBER, List.of(111,222,333,444,555,666,777), 7);
	}
}

File: programs-basic-kotlin/src/test/kotlin/com/local/numbertypes/NelsonNumberTest.kt
package com.local.numbertypes
import org.junit.jupiter.api.Test
class NelsonNumberTest : AbstractNumberTest() {
	@Test
	fun testNelsonNumbers() {
		testNumbersTypeByFile(com.local.numbertypes.NumberType.NELSON_NUMBER, "nelsonnumbers.txt")
	}

	@Test
	fun testNelsonNumbersByList() {
		// Pass the order as 5
		testNumbersTypeByList(com.local.numbertypes.NumberType.NELSON_NUMBER, java.util.List.of<Int>(111, 222, 333, 444, 555, 666, 777), 7)
	}
}